XRTK-Core/Packages/com.xrtk.core/Extensions/ArrayExtensions.cs

```diff
@@ -29,5 +29,20 @@ public static bool IsValidArray(this Array array)
         {
             return array != null && array.Length > 0;
         }
+
+        /// <summary>
+        /// Extends an existing array to add a new item
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="array">The array to extend</param>
+        /// <param name="newItem">The item to add to the array</param>
+        /// <returns></returns>
+        public static T[] AddItem<T>(this T[] array, T newItem)
```

Comment:
optional index specification?

might be better if pulled into a separate pr?

Reply:
Created #453.

Array.Copy (at least under Unity requires an index)